argument.go

```diff
@@ -501,6 +502,9 @@ func (o *arg) setDefaultFiles() error {
 func (o *arg) setDefault() error {
 	// Only set default if it was not parsed, and default value was defined
 	if !o.parsed && o.opts != nil && o.opts.Default != nil {
+		if o.noDefault {
+			return fmt.Errorf("argument [%s, %s] does not support default values", o.sname, o.lname)
+		}
```

Comment:
While this solution would fix the problem, I think it may cause issues with any existing code, which should be avoided (i am trying to maintain this library backward compatible as much as possible). If some code was setting it to `false` which would make it work as expected, with this change that code will start failing.

I think better options that maintain some sort of compatibility is to either ignore `Options.Default` for Flag or to add logic for inverting Flag value (`Options.Default` defaults to `false`, if set to `true` then if flag present the value set to false, otherwise true).

Reply:
I see what you mean. That logic would have to be handled in SetDefault() then. What do you think about adding an argType field to the arg interface that makes it easier to add type-specific functionality like this? I would think an enum would work really well, and I think that would not change any existing functionality.

Reply:
> What do you think about adding an argType field to the arg interface that makes it easier to introduce type-specific functionality like this?

Yes, I were considering something like that for V2, but current schedule does not allow me to start working on it just yet. My concern with the current one is just the backward compatibility, this lib is used by a number of tools, and would not want to introduce a breaking change here. So any solution you like that would not break existing code outright should be okay I think.